### PR TITLE
Updated plot_city

### LINKED ISSUE #2322: making the plot_city look nicer
### What is the expected enhancement?

- [ ] make rho the latex version
- [ ] use standard version of imag and real 
- [ ] set the size of real/imag to be the same for both plots (0.25, 0.5, 0.75, 1) depending on the largest element
- [ ] add the z axis to the bottom left 

image

### PR DESCRIPTION
**Pull request by dennis-liu-1**
### Summary

Fixes #2322 

### Details and comments

Matplotlib has a function called `get_autoscalez_on()` which automatically determines the best intervals to use. I used this in place of `set_major_locator(MaxNLocator(5))`, which would sometimes cut off the max and min values on the plot axes. Both the real/imag plots should have the same maximum value now.

### CODE REVIEW

File: qiskit/visualization/state_visualization.py [hunk 1]
```diff
@@ -318,40 +317,40 @@ def plot_state_city(rho, title="", figsize=None, color=None,
...
             ax2.set_zticks([0])
             eps = 1e-9
-            ax2.axes.set_zlim3d(np.min(dzi), np.max(dzi)+eps)
+            ax2.axes.set_zlim3d(np.min(dzi), max(np.max(dzr)+1e-9, np.max(dzi)+eps))
         else:
             ax2.axes.set_zlim3d(auto=True)
     ax2.w_xaxis.set_ticklabels(row_names, fontsize=14, rotation=45)
     ax2.w_yaxis.set_ticklabels(column_names, fontsize=14, rotation=-22.5)
-    ax2.set_zlabel("Imag[rho]", fontsize=14)
+    ax2.set_zlabel(r'Im[$\rho$]', fontsize=14)
```
Comment (1ucian0) [inline qiskit/visualization/state_visualization.py]:
```suggestion
    ax2.set_zlabel('Im[$\\rho$]', fontsize=14)
```

File: qiskit/visualization/state_visualization.py [hunk 2]
```diff
@@ -318,40 +317,40 @@ def plot_state_city(rho, title="", figsize=None, color=None,
...
-            ax1.axes.set_zlim3d(np.min(dzr), np.max(dzr)+1e-9)
+            ax1.axes.set_zlim3d(np.min(dzr), max(np.max(dzr)+1e-9, np.max(dzi)))
         else:
             ax1.axes.set_zlim3d(auto=True)
-    ax1.zaxis.set_major_locator(MaxNLocator(5))
+    ax1.get_autoscalez_on()
     ax1.w_xaxis.set_ticklabels(row_names, fontsize=14, rotation=45)
     ax1.w_yaxis.set_ticklabels(column_names, fontsize=14, rotation=-22.5)
-    ax1.set_zlabel("Real[rho]", fontsize=14)
+    ax1.set_zlabel(r'Re[$\rho$]', fontsize=14)
```
Comment (1ucian0) [inline qiskit/visualization/state_visualization.py]:
```suggestion
    ax1.set_zlabel('Re[$\\rho$]', fontsize=14)
```

**Comment by Travis-S-IBM** [comment]
@nonhermitian @mtreinish Is there anything else @dennis-liu-1 needs to do to enable this PR to be closed? If not, mind reviewing the changes?

**Comment by ajavadia** [comment]
do you mind pasting a picture of how the image will look like now? (so that we can contrast it with the image in #2322 )

**Comment by dennis-liu-1** [comment]
Screen Shot 2019-09-23 at 9 26 44 AM
The original plot for comparison, and another example as well

Screen Shot 2019-09-23 at 9 30 39 AM

**Comment by mtreinish** [comment]
Do we have a bug about the axis being off center on the imaginary half when there is a negative value? While that's outside the scope of this PR it's something else we'll want to fix.

**Comment by Travis-S-IBM** [comment]
@mtreinish There does not appear to be. As you note, it's beyond the scope of this PR. Can we please close it and then @dennis-liu-1 can work on the other issue.

File: qiskit/visualization/state_visualization.py [hunk 3]
```diff
@@ -318,40 +317,40 @@ def plot_state_city(rho, title="", figsize=None, color=None,
     ax1.set_yticks(np.arange(0.5, ly+0.5, 1))
     max_dzr = max(dzr)
     min_dzr = min(dzr)
+    min_dzi = np.min(dzi)
+    max_dzi = np.max(dzi)
     if max_dzr != min_dzr:
-        ax1.axes.set_zlim3d(np.min(dzr), np.max(dzr)+1e-9)
+        ax1.axes.set_zlim3d(np.min(dzr), max(np.max(dzr)+1e-9, np.max(dzi)))
```
Comment (mtreinish) [inline qiskit/visualization/state_visualization.py]:
```suggestion
        ax1.axes.set_zlim3d(np.min(dzr), max(np.max(dzr) + 1e-9, max_dzi))
```

File: qiskit/visualization/state_visualization.py [hunk 4]
```diff
@@ -318,40 +317,40 @@ def plot_state_city(rho, title="", figsize=None, color=None,
...
     else:
         if min_dzr == 0:
-            ax1.axes.set_zlim3d(np.min(dzr), np.max(dzr)+1e-9)
+            ax1.axes.set_zlim3d(np.min(dzr), max(np.max(dzr)+1e-9, np.max(dzi)))
```
Comment (mtreinish) [inline qiskit/visualization/state_visualization.py]:
```suggestion
            ax1.axes.set_zlim3d(np.min(dzr), max(np.max(dzr) + 1e-9, max_dzi))
```

File: qiskit/visualization/state_visualization.py [hunk 5]
```diff
@@ -318,40 +317,40 @@ def plot_state_city(rho, title="", figsize=None, color=None,
...
 
     ax2.set_xticks(np.arange(0.5, lx+0.5, 1))
     ax2.set_yticks(np.arange(0.5, ly+0.5, 1))
-    min_dzi = np.min(dzi)
-    max_dzi = np.max(dzi)
     if min_dzi != max_dzi:
         eps = 0
-        ax2.zaxis.set_major_locator(MaxNLocator(5))
-        ax2.axes.set_zlim3d(np.min(dzi), np.max(dzi)+eps)
+        ax2.axes.set_zlim3d(np.min(dzi), max(np.max(dzr)+1e-9, np.max(dzi)+eps))
```
Comment (mtreinish) [inline qiskit/visualization/state_visualization.py]:
```suggestion
        ax2.axes.set_zlim3d(min_dzi, max(np.max(dzr) + 1e-9, max_dzi + eps))
```

File: qiskit/visualization/state_visualization.py [hunk 6]
```diff
@@ -318,40 +317,40 @@ def plot_state_city(rho, title="", figsize=None, color=None,
...
     else:
         if min_dzi == 0:
             ax2.set_zticks([0])
             eps = 1e-9
-            ax2.axes.set_zlim3d(np.min(dzi), np.max(dzi)+eps)
+            ax2.axes.set_zlim3d(np.min(dzi), max(np.max(dzr)+1e-9, np.max(dzi)+eps))
```
Comment (mtreinish) [inline qiskit/visualization/state_visualization.py]:
```suggestion
            ax2.axes.set_zlim3d(min_dzi, max(np.max(dzr) + 1e-9, max_dzi + eps))
```

**Pull request merged**